### PR TITLE
Implements the operators: $regex, $text.

### PR DESCRIPTION
**Pull request by edumarcal**

### CODE REVIEW

File: binary_operator.go [hunk 1]
```diff
@@ -7,45 +7,60 @@ import (
...
+
+	// Element Operators
+	BinaryOperatorTypeExists
+	BinaryOperatorType
+
+	// Evaluation Operators
+	BinaryOperatorTypeRegex
+
+	// Others Operators
+	BinaryOperatorTypeLike
```
Comment (jamillosantos) [inline binary_operator.go]: Operators not in the default implementation of mongo. Check all that it applies.

File: operators_evaluation.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+type FindText struct {
+	Search             string  `bson:"$search"`
+	Language           *string `bson:"$language,omitempty"`
+	CaseSensitive      bool    `bson:"$caseSensitive,omitempty"`
+	DiacriticSensitive bool    `bson:"$diacriticSensitive,omitempty"`
+}
+
+func Text(value ... interface{}) *BooleanOperator {
```
Comment (jamillosantos) [inline operators_evaluation.go]: The only option for $text operator is to pass your `FindText` structure. So ask for it in the params. Do not ask for a "value...interface{}" ask for a *FindText pointer.

File: binary_operator.go (hunk 1 above)
Comment (edumarcal) [reply binary_operator.go]: OK, removed the constants of the operators' types does not default of mongo.

File: operators_evaluation.go (hunk 2 above)
Comment (edumarcal) [reply operators_evaluation.go]: Hum, I understand. I Changed the argument type to `Find Text` in function.

**Pull request merged**